src/core/SampleBuffer.cpp

```diff
@@ -90,6 +90,12 @@ SampleBuffer::SampleBuffer() :
 SampleBuffer::SampleBuffer(const QString & audioFile, bool isBase64Data)
 	: SampleBuffer()
 {
+	if (auto cachedBuffer = Engine::getSampleBufferCache()->get(audioFile.toStdString()); cachedBuffer.has_value())
+	{
+		*this = **cachedBuffer;
```

Comment:
Doing this will likely create crashes in certain cases because this assignment will also copy the following raw pointers in `SampleBuffer`:
```
sampleFrame * m_origData;
sampleFrame * m_data;
```
When either `this` or `cachedBuffer` is deleted it will release the memory allocated to `m_origData` and `m_data`. Therefore the other `SampleBuffer` will be left with invalid data leading to a crash when it tries to access it.